irohad/ametsuchi/block_storage.hpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BLOCK_STORAGE_HPP
+#define IROHA_BLOCK_STORAGE_HPP
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+#include <boost/optional.hpp>
+#include "interfaces/iroha_internal/block.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+
+    /**
+     * Append-only block storage interface
+     */
+    class BlockStorage {
+     public:
+      /**
+       * Type of storage key
+       */
+      using Identifier = uint32_t;
+
+      /**
+       * Append block with given key
+       * @return true if inserted successfully, false otherwise
+       */
+      virtual bool insert(
+          Identifier id,
+          std::shared_ptr<const shared_model::interface::Block> block) = 0;
+
+      [[deprecated("Use shared_ptr")]] virtual bool insert(
```

Comment:
Do we really need it?

Reply:
https://github.com/hyperledger/iroha/blob/a5acbdc1737c680c6d15a18c31c305ac3a7d16f6/irohad/ametsuchi/impl/mutable_storage_impl.cpp#L45
https://github.com/hyperledger/iroha/blob/a5acbdc1737c680c6d15a18c31c305ac3a7d16f6/irohad/ametsuchi/impl/mutable_storage_impl.cpp#L77
It can be removed when the parameter of this method is refactored to `shared_ptr`.